link-grammar/prune.c

```diff
@@ -573,14 +573,14 @@ static inline bool matches_S(connector_table *ct, Connector * c, char dir)
  */
 static int mark_dead_connectors(connector_table *ct, Exp * e, char dir)
 {
-	int count;
-	count = 0;
+	int count = 0;
+	Connector dummy;
+	init_connector(&dummy);
```

Comment:
why? you are changing the scope -- in general it is best to keep the scope closest to the location where it is actually used.   If dummy is not used anywhere else, then initialize it only in the block where it is used.